hathor/transaction/storage/transaction_storage.py

```diff
@@ -643,6 +655,40 @@ def is_running_manager(self) -> bool:
         """
         return self.get_value(self._manager_running_attribute) == '1'
 
+    def get_last_started_at(self) -> int:
+        """ Return the timestamp when the database was last started.
+        """
+        # XXX: defaults to 1 just to force indexes initialization, by being higher than 0
+        return int(self.get_value(self._last_start_attribute) or 1)
+
+    def set_last_started_at(self, timestamp: int) -> None:
+        """ Update the timestamp when the database was last started.
+        """
+        self.add_value(self._last_start_attribute, str(timestamp))
+
+    def get_index_last_started_at(self, index_db_name: str) -> int:
+        """ Return the timestamp when an index was last started.
+        """
+        return int(self.get_value(index_db_name) or 0)
+
+    def set_index_last_started_at(self, index_db_name: str, timestamp: int) -> None:
+        """ Update the timestamp when a specific index was last started.
+        """
+        self.add_value(index_db_name, str(timestamp))
+
+    def update_last_started_at(self, timestamp: int) -> None:
+        """ Updates the respective timestamps of when the node was last started.
+
+        Using this mehtod ensures that the same timestamp is being used and the correct indexes are being selected.
+        """
+        assert self.indexes is not None
+        self.set_last_started_at(timestamp)
+        for index in self.indexes.iter_all_indexes():
+            index_db_name = index.get_db_name()
+            if index_db_name is None:
+                continue
+            self.add_value(index_db_name, str(timestamp))
```

Comment:
Maybe add a prefix to avoid key collision?

Reply:
I think it makes sense, I'll move the "index" word from all the `get_db_name` to a common prefix, but not to here, this isn't a "centralized" place that controls the attribute name generation. Also, this is probably doing something that is already implemented somewhere else, so I'll refactor that too.

Reply:
Done.

Reply:
Thanks!